components/TextFields/src/MDCTextInputControllerBase.m

```diff
@@ -599,12 +601,16 @@ - (void)updateTextInput {
 - (void)updatePlaceholder {
   UIFont *placeHolderFont = self.inlinePlaceholderFont;
   if (self.mdc_adjustsFontForContentSizeCategory) {
+    // TODO: (#4331) This needs to be converted to the new text scheme.
     placeHolderFont =
         [placeHolderFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleBody1
                                       scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
 
   }
-  self.textInput.placeholderLabel.font = placeHolderFont;
+
+  if (![self.textInput.placeholderLabel.font isEqualToFont:placeHolderFont]) {
```

Comment:
Can you leave a comment here explaining why we have to do this?

Reply:
Done.